PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ConnectionStringCheck.java

```diff
@@ -0,0 +1,90 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.*;
+import com.intellij.psi.*;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONObject;
+
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This class extends the LocalInspectionTool and checks for the usage of connection strings API to create clients
+ * in the Azure SDK for Java.
+ * If a connection string is found, a warning is displayed.
+ */
+public class ConnectionStringCheck extends LocalInspectionTool {
+
+    private static final List <String> ruleConfigurations = loadRuleConfigurations();
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new JavaElementVisitor() {
+
+            @Override
+            public void visitElement(@NotNull PsiElement element) {
+                super.visitElement(element);
+
+                String METHOD_TO_CHECK = ruleConfigurations.get(0);
+                String SUGGESTION = ruleConfigurations.get(1);
+
+                // check if an element is an azure client
+                if (element instanceof PsiMethodCallExpression) {
+
+                    // check if the method is a connectionString call
+                    PsiMethodCallExpression methodCallExpression = (PsiMethodCallExpression) element;
+                    PsiReferenceExpression methodExpression = methodCallExpression.getMethodExpression();
+
+                    // resolvedMethod is the method that is being called
+                    PsiElement resolvedMethod = methodExpression.resolve();
+
+                    if (resolvedMethod != null && resolvedMethod instanceof PsiElement && ((PsiMethod) resolvedMethod).getName().equals(METHOD_TO_CHECK)) {
```

Comment:
Comment on L42 is actually applicable here.

Reply:
unresolved.

Reply:
resolved offline